services/api/src/models/billing.ts

```diff
@@ -0,0 +1,202 @@
+import moment from 'moment';
+import { Group, BillingGroup } from './group';
+import { getKeycloakAdminClient } from '../clients/keycloak-admin';
+import {
+  handleAddBillingModifier,
+  handleGetBillingGroupModifiers,
+  handleUpdateBillingModifier,
+  handleDeleteBillingModifier,
+  handleDeleteAllBillingGroupModifier
+} from '../resources/billing/helpers';
+
+import { GroupInput } from './group';
+
+export interface BillingModifier {
+  id?: number;
+  group?: Group | GroupInput;
+  startDate?: string;
+  endDate?: string;
+  discountFixed?: number;
+  discountPercentage?: number;
+  extraFixed?: number;
+  extraPercentage?: number;
+  customerComments?: string;
+  adminComments?: string;
+  weight?: number;
+}
+
+export interface BillingModifierInput extends BillingModifier {
+  group: GroupInput;
+}
+
+export interface AddBillingModifierInput {
+  input: BillingModifierInput;
+}
+
+export interface UpdateBillingModifierInput {
+  input: {
+    id: number;
+    patch: BillingModifier;
```

Comment:
This, and the other interfaces, are tightly coupling the model to the graphql API. If we ever wanted to use the model outside of a graphql resolver (like in a script, or maybe we need to sync billing groups to/from another system in the background), we'd need to create an object that resembles a graphql request first. And if the API ever changes, that might require changes to non-api sections.

I would prefer to keep the typings separate between the API and the model. See the `updateGroup` groups model and and resolvers for example. It's more verbose, but the resolver function I think should be the place where we translate API objects into the model objects.

Reply:
yeah, this was my attempt at strongly typing everything I was working on. I do think we should avoid "any" typing if possible. The code duplication between the API SDL and Typescript could be addressed with an ORM. 

I'll need to think about this one a bit more. The current convention and ultimate direction we're headed with Typescript isn't super clear to me yet.

Reply:
@rocketeerbkw I think this should be addressed in a6acb58b511b54795590fb88783d36066efc7caf